chips/earlgrey/src/chip.rs

```diff
@@ -461,39 +467,40 @@ core::arch::global_asm!(
     "
             .section .riscv.trap_vectored, \"ax\"
             .globl _start_trap_vectored
-          _start_trap_vectored:
-
-            j _start_trap
-            j _start_trap
-            j _start_trap
-            j _start_trap
-            j _start_trap
-            j _start_trap
-            j _start_trap
-            j _start_trap
-            j _start_trap
-            j _start_trap
-            j _start_trap
-            j _start_trap
-            j _start_trap
-            j _start_trap
-            j _start_trap
-            j _start_trap
-            j _start_trap
-            j _start_trap
-            j _start_trap
-            j _start_trap
-            j _start_trap
-            j _start_trap
-            j _start_trap
-            j _start_trap
-            j _start_trap
-            j _start_trap
-            j _start_trap
-            j _start_trap
-            j _start_trap
-            j _start_trap
-            j _start_trap
-            j _start_trap
-        "
+          _earlgrey_start_trap_vectored:
+
+            j {start_trap}
+            j {start_trap}
+            j {start_trap}
+            j {start_trap}
+            j {start_trap}
+            j {start_trap}
+            j {start_trap}
+            j {start_trap}
+            j {start_trap}
+            j {start_trap}
+            j {start_trap}
+            j {start_trap}
+            j {start_trap}
+            j {start_trap}
+            j {start_trap}
+            j {start_trap}
+            j {start_trap}
+            j {start_trap}
+            j {start_trap}
+            j {start_trap}
+            j {start_trap}
+            j {start_trap}
+            j {start_trap}
+            j {start_trap}
+            j {start_trap}
+            j {start_trap}
+            j {start_trap}
+            j {start_trap}
+            j {start_trap}
+            j {start_trap}
+            j {start_trap}
+            j {start_trap}
+    ",
+    start_trap = sym rv32i::_start_trap,
```

Comment:
These changes are tangential, but we should avoid referencing symbol names directly across crates. Instead, we should use proper Rust function exports and re-introduce them as locally-scoped symbols in Rust's `asm!` blocks.